rtengine/improcfun.cc

```diff
@@ -298,6 +298,18 @@ void ImProcFunctions::updateColorProfiles (const Glib::ustring& monitorProfile,
 #else
         monitor = ICCStore::getInstance()->getProfile (options.rtSettings.srgb);
 #endif
+//    }
+    Glib::ustring outtest = monitorProfile ;
+    
+    std::string fileis_RTv2 = outtest.substr(0, 4);
+
+    if((fileis_RTv2 == "RTv4" || fileis_RTv2 == "RTv2")  && gamutCheck) {
+            cmsToneCurve* GammaTRC[3];
+            GammaTRC[2] = cmsBuildGamma(NULL, 2.4);//if GammaTRC[2] used by v4 or v2 gamut check does not work...why ??
```

Comment:
I think we need to dig deeper into this, and understand exactly why it works... this is so odd that I suspect we are doing something not quite right here.